src/__tests__/Search.test.js

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import Search from "../components/Search";
+
+describe("Search", () => {
+  it("Renders Correctly", () => {
+    const { asFragment } = render(<Search />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it("Updates input value when typing", () => {
+    render(<Search />);
+    const inputElement = screen.getByPlaceholderText("Enter keywords...");
+
+    fireEvent.change(inputElement, {target: { value: "space" }} );
+    expect(inputElement.value).toBe("space");
+  });
+
+  it("Calls setSearchResults when form is submitted", async () => {
+    const setSearchResultsMock = jest.fn();
+    render(<Search setSearchResults={setSearchResultsMock} />);
+    const inputElement = screen.getByPlaceholderText("Enter keywords...");
+    const submitButton = screen.getByText("Go!");
+
+    fireEvent.change(inputElement, { target: { value: "moon" } });
+    fireEvent.click(submitButton);
+
+    await waitFor(() => {
+      expect(setSearchResultsMock).toHaveBeenCalledWith(expect.any(Array));
+    });
```

Comment:
the main problem with this test is you are actually making an api call to test that it works, as your function `handleSubmit` calls `getImages` before `setSearchResults`. To fix this, you need to mock `getImages`. You can do this as so:
```
import * as getImages from "../requests/getImages";
...

describe("Search", () => {
  ...

  it("Calls setSearchResults when form is submitted", async () => {
    const setSearchResultsMock = jest.fn();
    const mockReturn = jest.fn();
    jest.spyOn(getImages, "default").mockResolvedValue(mockReturn);

    render(<Search setSearchResults={setSearchResultsMock} />);
    const inputElement = screen.getByPlaceholderText("Enter keywords...");
    const submitButton = screen.getByText("Go!");

    const input = "moon"
    fireEvent.change(inputElement, { target: { value: input } });
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(getImages.default).toBeCalledWith(input)
    });

    await waitFor(() => {
      expect(setSearchResultsMock).toHaveBeenCalledWith(mockReturn);
    });
  });
});
```

Reply:
note: I have a separate test to expect to check if `getImages` is called with the input value. otherwise, this test will only test that pressing the `submitButton` calls `setSerachResults` with the output of `getImages`